daemon/checkpoint.go

```diff
@@ -8,6 +8,12 @@ import (
 	"path/filepath"
 
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/utils"
+)
+
+var (
+	validCheckpointNameChars   = utils.RestrictedNameChars
+	validCheckpointNamePattern = utils.RestrictedNamePattern
```

Comment:
I just realized `RestrictedNamePattern` is originally created for containers, and allows a leading `/` https://github.com/docker/docker/blob/b248de7e332b6e67b08a8981f68060e6ae629ccf/utils/names.go#L8-L9. 

So this is still "valid" with this change;

```bash
docker checkpoint create my-container /out-of-jail
```

The name is misleading, and so is the description. Perhaps we should rename that pattern to `RestrictedContainerNamePattern`

Or (maybe better); remove the leading `/` from that check, and change the check for container names here; https://github.com/docker/docker/blob/b248de7e332b6e67b08a8981f68060e6ae629ccf/daemon/names.go#L61

to

```go
if !validContainerNamePattern.MatchString(strings.TrimPrefix(name, "/')) {
   ...
}
```

If we do so, 
we can remove the  `RestrictedVolumeNamePattern` https://github.com/docker/docker/blob/b248de7e332b6e67b08a8981f68060e6ae629ccf/utils/names.go#L11-L12

And use `RestrictedNamePattern` as a default pattern for all objects

@tiborvass wdyt?